airflow/configuration.py

```diff
@@ -1307,13 +1307,25 @@ def has_option(self, section: str, option: str) -> bool:
         except (NoOptionError, NoSectionError):
             return False
 
+    def set(self, section: str, option: str, value: str | None = None) -> None:
+        """
+        Set an option to the given value.
+
+        This override just makes sure the section and option are lower case, to match what we do in `get`.
+        """
+        section = str(section).lower()
+        option = str(option).lower()
```

Comment:
Why do we need to do the additional `str()` casts?

Reply:
I copied what we do in `get` here
https://github.com/apache/airflow/blob/0fe6a607fc6f1678b678006fbea2ded068544f63/airflow/configuration.py#L953-L954
I suppose even though the type annotation is `str`, people can pass things that are not strings, and we want to avoid crashing in that case ?

Reply:
@uranusjr @Lee-W 
I think we should keep the `str()` to have a behavior consistent with `get`.
If I remove them, then `get("section", 12)` works but `set("section", 12, "value")` doesn't.

Reply:
Alternatively the `str()` in `get` should be removed as well. There’s zero use case to non-string keys, no existing keys would be affected.

Reply:
Yes. let's remove str's. We have no options that are non-strings. We can also mention it in newsfragment, but that's less necessary than mentioning casing change.